src/zmq_van.h

```diff
@@ -63,7 +63,8 @@ class ZMQVan : public Van {
     CHECK(receiver_ != NULL)
         << "create receiver socket failed: " << zmq_strerror(errno);
     int local = GetEnv("DMLC_LOCAL", 0);
-    std::string addr = local ? "ipc:///tmp/" : "tcp://*:";
+    std::string hostname = node.hostname.empty() ? "*" : node.hostname;
```

Comment:
where does hostname come from?

Reply:
It gets set here: https://github.com/dmlc/ps-lite/blob/master/src/van.cc#L65

Reply:
I'm confused. Is this local host? If you are listen only for localhost how can workers connect?
Have you tested this on multiple machines?

Reply:
This is tested on multiple machines on ec2.
Existing logic binds to a port on 0.0.0.0 - which is basically all interfaces on the host.
This change is not specific to localhost. ps-lite picks the first available non loopback interface if no ip is specified. This just ensures that we only listen on that one interface (IP) and not all of them.

Reply:
Does it log which interface it's listening on?

Reply:
Yes. It does. https://github.com/dmlc/ps-lite/blob/master/src/van.cc#L75
Today this log line is not accurate. For example it can say `Bind to role=scheduler, id=1, ip=127.0.0.1, port=9999, is_recovery=0` when in fact is listening on 0.0.0.0:9999

Reply:
I remember I tried "tcp://ip:port" before but it doesn't work. If @madjam verified that it's working now, probably due to we updated zeromq's version, then it should be ok

Reply:
This doc shows you can use with ip:port
http://api.zeromq.org/2-1:zmq-tcp
I don't know when they added/fixed this option.

As I mentioned, this change was tested by launching a distributed training job in ec2 using launch.py